dds/src/infrastructure/instance.rs

```diff
@@ -40,7 +40,7 @@ impl From<Guid> for InstanceHandle {
 
 impl From<InstanceHandle> for Guid {

```

Comment:
This implementation looks very ugly so I have to question if this direction of conversion is useful. In fact a Guid is guaranteed to be convertible into an instance handle but the other way around is not guaranteed. I think we should remove this conversion in a separate task and replace the "impl From<InstanceHandle> for [u8; 16]" by a AsRef following the same principle as the other types

Reply:
Absolutely. The intent to remove this conversion was the starting point for this PR. As I started I realized that the implementations of the submessage elements need to be cleaned first.